### PR TITLE
LUCENE-10495: Fix return statement of siblingsLoaded() in TaxonomyFacets

### PR DESCRIPTION
**Pull request by Yuti-G**
# Description

Found a bug in TaxonomyFacets when trying to use the siblingsLoaded function. Currently, siblingsLoaded() returns children!= null, but it should return siblings != null;

# Solution
Fixed the return statement: return siblings != null.

# Tests
Added test case testSiblingsLoaded() in TestTaxonomyFacetSumValueSource where siblingsLoaded() and childrenLoaded() can return different boolean values. 

# Checklist

Please review the following and check all that apply:

- [X] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/lucene/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [X] I have created a Jira issue and added the issue ID to my pull request title.
- [X] I have given Lucene maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [X] I have developed this patch against the `main` branch.
- [X] I have run `./gradlew check`.
- [X] I have added tests for my changes.

### CODE REVIEW

File: lucene/facet/src/java/org/apache/lucene/facet/taxonomy/TaxonomyFacets.java [hunk 1]
```diff
@@ -109,7 +109,7 @@ public boolean childrenLoaded() {
    * @lucene.experimental
    */
   public boolean siblingsLoaded() {
-    return children != null;
+    return siblings != null;
```
[the remark this answers is not in the document]
Comment (Yuti-G) [reply lucene/facet/src/java/org/apache/lucene/facet/taxonomy/TaxonomyFacets.java:112]: Hi @gsmiller, thanks for your feedback! There is only one use case that I could find where `siblingsLoaded()` and `childrenLoaded()` can return different boolean value, and I added a test for it. Please let me know if there is any question. Thanks again!

[the remark this answers is not in the document]
Comment (gsmiller) [reply lucene/facet/src/java/org/apache/lucene/facet/taxonomy/TaxonomyFacets.java:112]: Thanks @Yuti-G . Looks good!

**Comment by gsmiller** [comment]
@Yuti-G looks like a conflict needs to be resolved when you get a chance. Also, could you please add an entry to CHANGES.txt noting the bug fix please? We'll backport this to 9.2, so it would make sense to add it in the 9.2 section. Thanks again for finding this and fixing it!

**Comment by Yuti-G** [comment]
Thanks @gsmiller! I resolved the conflicts and added an entry to CHANGES.txt.

**Review by zhaih — APPROVED** [review]
LGTM, I'll just merge it to avoid further conflicts, please remember to create a backport PR to 9.x branch @Yuti-G 
Thank you!

**Comment by Yuti-G** [comment]
Thanks @zhaih !

**Pull request merged**